microprofile.ls/README.md

```diff
@@ -43,8 +45,52 @@ Building the Language Server
 Clients
 -------
 
-Here are some clients consuming the Quarkus language server:
+Here are some clients consuming the MicroProfile language server:
 
  * Eclipse with [quarkus-lsp4e (POC)](https://github.com/angelozerr/quarkus-lsp4e)
  * IntelliJ with [intellij-quarkus](https://github.com/jeffmaury/intellij-quarkus)
- * Visual Studio Code with [vscode-quarkus](https://github.com/redhat-developer/vscode-quarkus)
\ No newline at end of file
+ * Visual Studio Code with [vscode-quarkus](https://github.com/redhat-developer/vscode-quarkus)
+ 
+Code Snippets
+-------
+
+Java and properties completion snippets are managed by the MicroProfile LS (snippets on server side) by using Java SPI.
+
+## Describing snippets in JSON
+
+`Snippets` are described in JSON files using the [vscode snippet format](https://code.visualstudio.com/docs/editor/userdefinedsnippets#_create-your-own-snippets).
+
+The `"context"` key can be provided to specify the condition in which the snippet should appear:
+
+ * for `properties files`: the context/extension can be declared in the snippet to show the snippet only if the extension belongs to the project.
+
+```json
+"Add datasource properties": {
+		"prefix": "qds",
+		"body": [
+			...
+		],
+		"context": {
+			"extension": "quarkus-agroal"
+		}
+	}
+```
+
+means that the snippet is shown only if the project has the `quarkus-agroal` dependency.
+    
+ * for `Java files` (not available for the moment)
+ 
+## Adding new internal snippets
+
+To register a snippet you need to add it in :
```

Comment:
To register a snippet, it must be added in:

Reply:
fixed